proposals/lambda-method-group-defaults.md

```diff
@@ -134,42 +169,82 @@ DoAction(writeInt, 3); // Error, cannot convert from anonymous delegate type to
 void DoAction(Action<int> a, int p) {
   a(p);
 }
+
+int Count(params int[] xs) {
+  return xs.Length;
+}
+var counter = Count; // Inferred as anonymous delegate type
+DoFunction(counter, 3); // Error, cannot convert from anonymous delegate type to Func
+
+int DoFunction(Func<int[], int> f, int p) {
+  return f(new[] { p });
+}
 ```
 The impact of this breaking change needs to be considered. Fortunately, the use of `var` to infer the type of a method group has
-only been supported since C# 10, so only code which has been written since then which explicity relies on this behavior would break.
+only been supported since C# 10, so only code which has been written since then which explicitly relies on this behavior would break.
 
 ## Detailed design
 
 [design]: #detailed-design
 
-### Grammar and Parser Changes
+### Grammar and parser changes
 This enhancement requires the following changes to the grammar for lambda expressions.
 ```diff
-explicit_anonymous_function_parameter
+ explicit_anonymous_function_parameter_list
++    : explicit_anonymous_function_parameters (',' parameter_array)?
++    | parameter_array
++    ;
```

Comment:
The grammar changes imply that we support `params` and optional parameters in anonymous methods. Essentially `delegate (params int[] a)`. I don't see any examples of this in the spec though. Is it intended to be allowed?

Reply:
Yes, I think that lambdas and anonymous methods are the same in this regard. I don't know if that was the intention. But if yes, I can mention `delegate () {}` anonymous methods in the proposal...

Reply:
I changed the grammar to disallow these improvements in `delegate { }` syntax: 8585a197713535978d072b975737835fe149a03f

I based my syntax on [the one from `lambda-improvements.md`](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/lambda-improvements.md#syntax). But I don't quite understand why it mentions `equals_value_clause`, it's not defined anywhere and seems unused in that proposal. So, in my diff of the syntax, I act as if it wasn't there...